packages/itwinui-react/src/core/Select/Select.tsx

```diff
@@ -228,249 +228,249 @@ export type SelectProps<T> = {
-    const selectRef = React.useRef<HTMLDivElement>(null);
-
-    const show = React.useCallback(() => {
-      if (disabled) {
-        return;
-      }
-      setIsOpen(true);
-      popoverProps?.onVisibleChange?.(true);
-    }, [disabled, popoverProps]);
-
-    const hide = React.useCallback(() => {
-      setIsOpen(false);
-      selectRef.current?.focus({ preventScroll: true }); // move focus back to select button
-      popoverProps?.onVisibleChange?.(false);
-    }, [popoverProps]);
-
-    const menuItems = React.useMemo(() => {
-      return options.map((option, index) => {
-        const isSelected = isMultipleEnabled(value, multiple)
-          ? value?.includes(option.value) ?? false
-          : value === option.value;
-        const menuItem: JSX.Element = itemRenderer ? (
-          itemRenderer(option, { close: () => setIsOpen(false), isSelected })
-        ) : (
-          <MenuItem>{option.label}</MenuItem>
-        );
-
-        const { label, icon, startIcon: startIconProp, ...restOption } = option;
-
-        const startIcon = startIconProp ?? icon;
-
-        return React.cloneElement<MenuItemProps>(menuItem, {
-          key: `${label}-${index}`,
-          isSelected,
-          startIcon: startIcon,
-          endIcon: isSelected ? <SvgCheckmark aria-hidden /> : null,
-          onClick: () => {
-            if (option.disabled) {
-              return;
-            }
-
-            // update internal value state and also call external onChange
-            if (isSingleOnChange(onChangeRef.current, multiple)) {
-              setUncontrolledValue(option.value);
-              onChangeRef.current?.(option.value);
-              hide();
-            } else {
-              setUncontrolledValue((prev: T[]) =>
-                isSelected
-                  ? prev?.filter((i) => option.value !== i)
-                  : [...(prev ?? []), option.value],
-              );
-              onChangeRef.current?.(
-                option.value,
-                isSelected ? 'removed' : 'added',
-              );
-            }
-
-            // update live region
-            if (isMultipleEnabled(value, multiple)) {
-              const prevSelectedValue = value || [];
-              const newSelectedValue = isSelected
-                ? prevSelectedValue.filter((i) => option.value !== i)
-                : [...prevSelectedValue, option.value];
-              setLiveRegionSelection(
-                options
-                  .filter((i) => newSelectedValue.includes(i.value))
-                  .map((item) => item.label)
-                  .filter(Boolean)
-                  .join(', '),
-              );
-            }
-          },
-          ref: (el: HTMLElement) => {
-            if (isSelected && !multiple) {
-              el?.scrollIntoView({ block: 'nearest' });
-            }
-          },
-          role: 'option',
-          ...restOption,
-          ...menuItem.props,
-        });
+
+const SelectInnerFn = <T,>(
```

Comment:
i think you shouldn't need this inner function, you can revert the whole thing and add the type assertion at the end of the original component

```jsx
export const Select = React.forwardRef(
  <T,>(props, forwardedRef) => {
    // …
  }
) as <T>(
  props: SelectProps<T> & { ref?: React.ForwardedRef<HTMLElement> },
) => JSX.Element;
```